araproc/framework/map_utilities.py

```diff
@@ -1,74 +1,296 @@
     ----------
-    station_id : int
-        ID of the station from which to retrieve antenna information.
-    num_channels : int
-        Number of channels (antennas) at the station.
+    - station_id (int): ID of the station.
 
     Returns
     -------
-    avg_x : float
-        Average x-coordinate of the antennas at the specified station.
-    avg_y : float
-        Average y-coordinate of the antennas at the specified station.
-    avg_z : float
-        Average z-coordinate of the antennas at the specified station.
+    - Object with dynamically bound functions for use.
     """
-    geom_tool = ROOT.AraGeomTool.Instance()
-    
-    # Initialize sums for x, y, and z coordinates
-    sum_x, sum_y, sum_z = 0.0, 0.0, 0.0
-    
-    # Sum up x, y, and z coordinates for all antennas
-    for ant in range(num_channels):
-        ant_info = geom_tool.getStationInfo(station_id).getAntennaInfo(ant).antLocation
-        sum_x += ant_info[0]
-        sum_y += ant_info[1]
-        sum_z += ant_info[2]
-    
-    # Calculate averages
-    avg_x = sum_x / num_channels
-    avg_y = sum_y / num_channels
-    avg_z = sum_z / num_channels
-    
-    return avg_x, avg_y, avg_z
+    # Load required ARA libraries
+    ROOT.gSystem.Load(os.environ.get('ARA_UTIL_INSTALL_DIR') + "/lib/libAraEvent.so")
+
+    # Constants
+    FEET_TO_METER = 0.3048  # Conversion factor from feet to meter
+    #translator vector from the Surveyers Coordinate System to the ARA Global Coordinate System
+    SUR_TO_GLOB = np.array([22400, 53900, -36.02])  # These are in feets
+
+    # Initialize the Geometry Tool
+    geomTool = ROOT.AraGeomTool.Instance()
+
+    # Initialize the station informations
+    st_info = geomTool.getStationInfo(station_id)
+
+    # Define all functions
+    def get_station_center(excluded_channels=None):
+        """
+        Calculates the average x, y, and z coordinates of antennas for a given station.
+
+        Parameters
+        ----------
+        excluded_channels : list
+           list of channels (antennas) to exclude.
+
+        Returns
+        -------
+        av_ant_position:tuple
+           station center (x,y,z) for given station 
+        """ 
+        if excluded_channels is None:
+            excluded_channels = []
+
+        antenna_coordinates = [[] for _ in range(3)]  # x, y, z
+        total_antennas = constants.num_RF_channels
+
+        for antenna_index in range(total_antennas):
+            if antenna_index in excluded_channels:
+                continue
+
+            antenna_info = st_info.getAntennaInfo(antenna_index)
+            for axis in range(3):
+                antenna_coordinates[axis].append(antenna_info.antLocation[axis])
+        av_ant_position = tuple(np.average(coords) for coords in antenna_coordinates)
+        return av_ant_position
+
+    def get_local_CP(cal_pulser_index):
+        """
+        Parameters
+        ----------
+        cal_pulser_index: int 
+            The index of calpulser (0-3) 
```

Comment:
Maybe some more description would be helpful here. I'm guess 2 of these are the Vpols and 2 are the Hpols? Can you say which are which?